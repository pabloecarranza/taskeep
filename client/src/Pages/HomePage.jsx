import React from 'react';
import { Sidebar } from './../components/Sidebar';
import { TaskArea } from './../components/TaskArea';
import { Box, Flex } from '@chakra-ui/react';

export const HomePage = () => {
	return (
		<Flex justifyContent='center' mt='5px'>
			<Sidebar />
			<TaskArea />
		</Flex>
	);
};
