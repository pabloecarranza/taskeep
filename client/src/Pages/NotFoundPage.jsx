import {
	Heading,
	Container,
	Box,
	Center,
	Button,
	Text,
	Image,
} from '@chakra-ui/react';
import Logo from '../assets/daily-tasks.png';
import { ArrowForwardIcon } from '@chakra-ui/icons';
import { useNavigate } from 'react-router-dom';
import PropTypes from 'prop-types';

export const NotFoundPage = ({
	error_text,
	message_redirect,
	page_redirect,
	test = false,
}) => {
	let navigate = useNavigate();
	const reDirectTo = () => {
		navigate('/homepage', { replace: true });
	};

	return (
		<Container centerContent>
			<Center padding='4' w='100vw' h='100vh'>
				<Box
					w='80%'
					h='80%'
					display='flex'
					flexDir='column'
					alignItems='center'
					justifyContent='center'
				>
					<Box>
						<Image src={Logo} alt='Logo_Task' boxSize='150px' />
					</Box>
					<Heading as='h2' size='2xl' fontWeight='bold' textAlign='center'>
						{error_text}
						<Text
							bgGradient={test ? '' : 'linear(to-l, #7928ca, #0084ff)'}
							bgClip='text'
							fontSize='3xl'
							fontWeight='bold'
							pb='30px'
							pt='20px'
						>
							{message_redirect}
						</Text>
					</Heading>
					<Box>
						<Button
							rightIcon={<ArrowForwardIcon />}
							colorScheme='blue'
							size='lg'
							onClick={reDirectTo}
						>
							{page_redirect}
						</Button>
					</Box>
				</Box>
			</Center>
		</Container>
	);
};

NotFoundPage.propTypes = {
	error_text: PropTypes.string.isRequired,
	message_redirect: PropTypes.string.isRequired,
	page_redirect: PropTypes.string.isRequired,
};
