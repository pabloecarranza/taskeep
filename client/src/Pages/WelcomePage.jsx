import {
	Box,
	Flex,
	Text,
	Heading,
	Button,
	Center,
	Image,
	useDisclosure,
	ScaleFade,
	Link,
} from '@chakra-ui/react';
import { ExternalLinkIcon } from '@chakra-ui/icons';
import { BsLinkedin, VscGithubInverted } from 'react-icons/all';
import Logo from '../assets/daily-tasks.png';
import Typewriter from 'typewriter-effect';
import { LoginForm } from '../components/LoginForm';
import { ArrowForwardIcon } from '@chakra-ui/icons';

import bg from '../assets/hojas.png';

export const WelcomePage = () => {
	const { isOpen, onToggle } = useDisclosure();

	return (
		<Flex
			w='100vw'
			h='100vh'
			flexDir={{ base: 'column', md: 'row', lg: 'row' }}
		>
			<Box w='100%' h='100%'>
				<Center
					w='100%'
					h='100%'
					flexDir='column'
					alignItems='flex-end'
					textAlign='right'
					pr='35px'
				>
					<Box>
						<Image src={Logo} alt='Logo_Task' boxSize='150px' />
					</Box>
					<Text
						bgGradient='linear(to-l, #b785e9, #61acf3)'
						bgClip='text'
						fontSize='6xl'
						fontWeight='extrabold'
						pb='10px'
					>
						TASKEEP
					</Text>
					<Heading as='h2' size='2xl' noOflines={1} fontWeight='bold'>
						<Typewriter
							options={{
								strings: ['The most powerful', 'The real powerful'],
								autoStart: true,
								loop: true,
							}}
						/>

						<Text
							bgGradient='linear(to-l, #7928ca, #0084ff)'
							bgClip='text'
							fontSize='5xl'
							fontWeight='bold'
							pb='50px'
						>
							Task Management App
						</Text>
					</Heading>
					<Box>
						<Button
							size='lg'
							colorScheme='blue'
							rightIcon={<VscGithubInverted />}
							mr='10px'
							onClick={() =>
								window.open(
									'https://github.com/pabloecarranza/taskeep',
									'_blank'
								)
							}
						>
							GitHub
						</Button>
						<Button
							rightIcon={<ArrowForwardIcon />}
							colorScheme='blue'
							variant='outline'
							size='lg'
							onClick={onToggle}
						>
							Get Started
						</Button>
					</Box>

					<Button
						colorScheme='gray'
						variant='link'
						rightIcon={<BsLinkedin />}
						pt='35px'
						onClick={() =>
							window.open(
								'https://www.linkedin.com/in/pabloecarranza/',
								'_blank'
							)
						}
					>
						Made by Pablo Carranza
					</Button>
				</Center>
			</Box>
			<Center
				w='80%'
				h='100%'
				backgroundImage={bg}
				backgroundPosition='center'
				backgroundRepeat='no-repeat'
				backgroundSize='cover'
			>
				<ScaleFade initialScale={0.9} in={isOpen}>
					<LoginForm />
				</ScaleFade>
			</Center>
		</Flex>
	);
};
