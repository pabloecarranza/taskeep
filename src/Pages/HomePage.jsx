import { Flex } from '@chakra-ui/react';
import { useEffect } from 'react';
import { Sidebar } from '../components/Sidebar/Sidebar';
import { MainArea } from './../components/Routes/MainArea';
import { useDispatch } from 'react-redux';
import { sessionIn, sessionOut } from '../features/api/sessionSlice';
import { useNavigate } from 'react-router-dom';
export const HomePage = () => {
	const userData = JSON.parse(localStorage.getItem('identified-user'));
	const dispatch = useDispatch();
	const navigate = useNavigate();
	const userCheck = () => {
		if (!userData) {
			dispatch(sessionOut());
			navigate('/');
			return;
		}
		dispatch(sessionIn(userData));
	};

	useEffect(() => {
		userCheck();
	}, []);

	return (
		<Flex justifyContent='center' mt='5px'>
			<Sidebar />
			<MainArea />
		</Flex>
	);
};
