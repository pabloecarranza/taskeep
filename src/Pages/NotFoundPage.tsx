import {
	Heading,
	Container,
	Box,
	Center,
	Button,
	Text,
	Image,
} from '@chakra-ui/react';
import Logo from '../assets/daily-tasks.png';
import { ArrowForwardIcon } from '@chakra-ui/icons';
import { useNavigate } from 'react-router-dom';
import PropTypes from 'prop-types';

interface Props {
	errorText: string,
	messageRedirect: string,
	pageRedirect: string,
	test: boolean,
}

export const NotFoundPage = ({
	errorText,
	messageRedirect,
	pageRedirect,
	test = false,
}: Props ) => {
	const navigate = useNavigate();
	const reDirectTo = () => {
		navigate('/', { replace: true });
	};

	return (
		<Container centerContent>
			<Center padding='4' w='100vw' h='100vh'>
				<Box
					w='80%'
					h='80%'
					display='flex'
					flexDir='column'
					alignItems='center'
					justifyContent='center'
				>
					<Box>
						<Image src={Logo} alt='Logo_Task' boxSize='150px' />
					</Box>
					<Heading as='h2' size='2xl' fontWeight='bold' textAlign='center'>
						{errorText}
						<Text
							bgGradient={test ? '' : 'linear(to-l, #7928ca, #0084ff)'}
							bgClip='text'
							fontSize='3xl'
							fontWeight='bold'
							pb='30px'
							pt='20px'
						>
							{messageRedirect}
						</Text>
					</Heading>
					<Box>
						<Button
							rightIcon={<ArrowForwardIcon />}
							colorScheme='blue'
							size='lg'
							onClick={reDirectTo}
						>
							{pageRedirect}
						</Button>
					</Box>
				</Box>
			</Center>
		</Container>
	);
};

NotFoundPage.propTypes = {
	errorText: PropTypes.string.isRequired,
	messageRedirect: PropTypes.string.isRequired,
	pageRedirect: PropTypes.string.isRequired,
	test: PropTypes.bool,
};
