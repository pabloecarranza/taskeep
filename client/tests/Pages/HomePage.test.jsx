import { render, screen } from '@testing-library/react';
import { vi, expect, describe, it, test } from 'vitest';
import { HomePage } from './../../src/Pages/HomePage';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Name of the group', () => {
	vi.mock('../../src/components/Sidebar/Sidebar', () => {
		return {
			__esModule: true,
			default: () => {
				return <Sidebar />;
			},
			Sidebar: () => {
				return <div />;
			},
		};
	});

	vi.mock('../../src/components/TaskArea.jsx', () => {
		return {
			__esModule: true,
			default: () => {
				return <TaskArea />;
			},
			TaskArea: () => {
				return <div />;
			},
		};
	});

	it('should match with snapshot', () => {
		const { container } = render(
			<Router>
				<HomePage />
			</Router>
		);

		expect(container).toMatchSnapshot();
	});
});
