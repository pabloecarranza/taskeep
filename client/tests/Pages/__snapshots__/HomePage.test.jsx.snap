// Vitest Snapshot v1

exports[`Name of the group > should  1`] = `
<div>
  <div
    class="css-ctfmb8"
  >
    <div />
    <div />
  </div>
</div>
`;
